src/Components/WebAssembly/testassets/ThreadingApp/Pages/Counter.razor

```diff
@@ -0,0 +1,37 @@
+﻿@page "/counter"
+@using System.Runtime.InteropServices
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    int currentCount = 0;
+
+    void IncrementCount()
+    {
+        currentCount++;
+    }
+
+    protected override Task OnInitializedAsync()
+    {
+        if(!RuntimeInformation.IsOSPlatform(OSPlatform.Create("Browser"))){
+            return Task.CompletedTask;
+        }
+        new Timer(async (state) =>
+        {
+            // send me to the thread pool
+            await Task.Delay(10).ConfigureAwait(false);
+
+            await InvokeAsync(() =>
+            {
+                // we are back on main thread
```

Comment:
Would it be useful to check here (for the test) that we are indeed "back on main thread"? Or will one of the following method calls fail if it's not?

Reply:
It would fail otherwise, the other threads don't have DOM access nor any JSInterop and runtime interop has asserts for it.

Reply:
I added few more asserts, thanks